src/DotNetLightning.Core/Channel/CommitmentsModule.fs

```diff
@@ -437,6 +438,106 @@ module RemoteForceClose =
         return obscuredCommitmentNumber
     }
 
+    let createPunishmentTx (perCommitmentSecret: PerCommitmentSecret)
+                           (commitments: Commitments)
+                           (localChannelPrivKeys: ChannelPrivKeys)
+                           (network: Network) 
+                               : TransactionBuilder =
+
+        let localChannelPubKeys = commitments.LocalParams.ChannelPubKeys
+        let remoteChannelPubKeys = commitments.RemoteParams.ChannelPubKeys
+
+        let perCommitmentPoint = perCommitmentSecret.PerCommitmentPoint()
+
+        let localCommitmentPubKeys =
+            perCommitmentPoint.DeriveCommitmentPubKeys localChannelPubKeys
+
+        let remoteCommitmentPubKeys =
+            perCommitmentPoint.DeriveCommitmentPubKeys remoteChannelPubKeys
+
+        let transactionBuilder = network.CreateTransactionBuilder()
```

Comment:
There's a function in `Transactions` called `createTransactionBuilder` which also disables input/output shuffling on the `TransactionBuilder`. You need to either call that or disable shuffling here.

Actually a better option would be to add a method to `Network` in the `NBitcoinExtensions` module called `CreateTransactionBuilderNoShuffling` or something like that.